### DISCUSSION TITLE
Lecture Reorganisation to Topic Series

### DISCUSSION THREAD

**Discussion by mmcky** [QuantEcon/meta#111] [General]
This discussion thread is to resolve some of the metadata that is outstanding for the new lecture reorganisation work as we move from `difficulty` based content to `topic` based content. Further details are available on the `notion` project page. 

Reviewing the metadata and names upfront will enable us to build the infrastructure more easily (i.e. urls, route53, GitHub actions, and repos)

The following details are suggested starting points and open to discussion. 

@jstac @thomassargent30 will have the final say on the decision column. 

## Statistics (25)

| Item | Option 1 | Option 2 | Decision | Approved |
|------|---------|----------|----------|----------|
| Title  | Statistics for Computational Economics | | Statistics for Computational Economics  | 🆗   |
| url | stats.quantecon.org | statistics.quantecon.org  | stats.quantecon.org | 🆗   |
| repo | lecture-stats | |  lecture-stats | 🆗  |
| GA4 | **G-NJV5XPRDFV**

## Dynamic Linear Economies (25)

| Item | Option 1 | Option 2 | Decision | Approved |
|------|---------|----------|----------|----------|
| Title  | Dynamic Linear Economies | | Dynamic Linear Economies  | 🆗   |
| url | dle.quantecon.org | dyn-lin-econ.quantecon.org  | dle.quantecon.org | 🆗   |
| repo | lecture-dle | | lecture-dle | 🆗   |
| GA4 | **G-D5TQMHKC2V**

## Dynamic Programming Squared (9)

| Item | Option 1 | Option 2 | Decision | Approved |
|------|---------|----------|----------|----------|
| Title  | Advanced Dynamic Programming  | | Dynamic Programming Squared  | 🆗   |
| url | dyn-prog-adv.quantecon.org | dps.quantecon.org  | dps.quantecon.org | 🆗   |
| repo | lecture-dyn-prog-adv | |  lecture-dps | 🆗   |
| GA4 | **G-E8SYFS4JB4**

## Equilibrium Models (14)

| Item | Option 1 | Option 2 | Decision | Approved |
|------|---------|----------|----------|----------|
| Title  | Equilibrium Models  | | Equilibrium Models  | 🆗   |
| url | eq-models.quantecon.org | equilibrium-models.quantecon.org  | eqm.quantecon.org |  🆗   |
| repo | lecture-equilibrium-models | | lecture-eqm | 🆗   |
| GA4 | **G-PEBDD57BFH**

## Introduction to Dynamics (30)

| Item | Option 1 | Option 2 | Decision | Approved |
|------|---------|----------|----------|----------|
| Title  | Introduction to Economic Dynamics  | | Introduction to Economic Dynamics  | 🆗   |
| url | dynamics-intro.quantecon.org | intro-dynamics.quantecon.org  | dynamics.quantecon.org | 🆗   |
| repo | lecture-intro-dynamics | | lecture-dynamics | 🆗   |
| GA4 | **G-Y9JY7E8Z7N**

## Tools and Techniques (15)

| Item | Option 1 | Option 2 | Decision | Approved |
|------|---------|----------|----------|----------|
| Title  | Tools & Techniques for Computational Economics  | | Tools & Techniques for Computational Economics  | 🆗   |
| url | tools-techniques.quantecon.org | tools.quantecon.org  | tools-techniques.quantecon.org | 🆗   |
| repo | lecture-tools-techniques | |  lecture-tools-techniques | 🆗   |
| GA4 | **G-FH5JZQ6T4G**

**Note:** this section will now include the `Time Series Models` lectures

**Comment by HumphreyYang** [comment]
Hi @mmcky,

The reorganized lectures are now hosted in the pages below:

https://quantecon.github.io/lecture-dle/intro.html
https://quantecon.github.io/lecture-dps/intro.html
https://quantecon.github.io/lecture-eqm/intro.html
https://quantecon.github.io/lecture-stats/intro.html
https://quantecon.github.io/lecture-tools-techniques/intro.html
https://quantecon.github.io/lecture-dynamics/intro.html

I quickly looked over at the builds and did not spot any issues other than the series titles, which are to be decided in our next meeting. Please let me know if you find anything strange.

The updated repositories are located in:

https://github.com/QuantEcon/lecture-dle
https://github.com/QuantEcon/lecture-dps
https://github.com/QuantEcon/lecture-eqm
https://github.com/QuantEcon/lecture-stats
https://github.com/QuantEcon/lecture-tools-techniques
https://github.com/QuantEcon/lecture-dynamics

**Update 01st Nov:** updated links to new repositories.

**Reply by mmcky** [reply]
Thanks @HumphreyYang this is great 👍 

@jstac and @thomassargent30 these links are just temporary so we can use them in the meeting to see how the new `topic` series will look.

**Reply by mmcky** [reply]
@HumphreyYang I have update the gh-pages hosting links for all new repo names

**Comment by jstac** [comment]
For this one: https://quantecon.github.io/reorg.dynamic-linear-economies/intro.html

Please:

- Change the title of the lecture series to the new one
- Put the Kalman filter section first --- and I guess we should add Tom's new lecture on the Kalman filter?
- Put the LQ Control section next
- Put the Dynamic Linear Economies section next
- See https://github.com/QuantEcon/meta/issues/113
- In "This website presents a set of advanced lectures on quantitative economic modeling, designed and written by [Thomas J. Sargent](http://www.tomsargent.com/) and [John Stachurski](http://johnstachurski.net/)." change "advanced lectures on quantitative economic modeling" to "lectures on dynamic linear economies"

**Comment by jstac** [comment]
For this one: https://quantecon.github.io/reorg.dynamic-programming-squared/intro.html

Please

- Change the title of the lecture series to the new one
- ask @thomassargent30 if there is a way to break the lectures into two or three sections, with the section title being different to the lecture series title (which is "Dynamic Programming Squared").
- See https://github.com/QuantEcon/meta/issues/113
- Change "advanced lectures on quantitative economic modeling" on the top page to "lectures on a modeling technique called dynamic programming squared"

**Comment by jstac** [comment]
For this one: https://quantecon.github.io/reorg.equilibrium-models/intro.html

- Change title
- Change section title "Rational Expectations Model" to "Rational Expectation Models"
- Change "advanced lectures on quantitative economic modeling" on the top page to "equilibrium economic models"
- See https://github.com/QuantEcon/meta/issues/113

**Comment by jstac** [comment]
Regarding https://quantecon.github.io/reorg.statistics/intro.html

- Change title
- Change "advanced lectures on quantitative economic modeling" on the top page to "statistical methods"
- Shift lectures 21, 22, 23 to the top of "Elementary Statistics"
- Lecture "[Job Search VII: Search with Learning](https://quantecon.github.io/reorg.statistics/odu.html)" should be shifted to the "Search" section of https://quantecon.github.io/reorg.introduction-dynamics/intro.html and the title adjusted accordingly (since there would be two VII lectures --- I suggest this one goes first).

**Comment by jstac** [comment]
Regarding https://quantecon.github.io/reorg.introduction-dynamics/intro.html

- Add a new section title before lecture 6 called "Optimal Savings" 
- Change title
- Change "advanced lectures on quantitative economic modeling" on the top page to "economic dynamics"

**Comment by jstac** [comment]
The lecture series https://quantecon.github.io/reorg.time-series-models/intro.html is too small.  I suggest that we fold it into https://quantecon.github.io/reorg.tools-techniques/intro.html as a new section, titled "Time Series".

Regarding https://quantecon.github.io/reorg.tools-techniques/intro.html,

- Change title
- Change "advanced lectures on quantitative economic modeling"  appropriately
- Change section title "Tools & Techniques" to "Solution Methods"  --- no section title should be the same as the series title.
- Section title "Risk Model Uncertainty and Robustness" needs a comma after risk.  Also, this section contains the hardest material so should be shifted to the end.

**Comment by HumphreyYang** [comment]
Hi @jstac,

Many thanks for your comments. I will address them this week.

I did not change the series titles because they have not been discussed. Should I use the titles listed in the table?

Many thanks in advance.

**Reply by jstac** [reply]
Thanks @HumphreyYang , and yes please (regarding titles).

**Comment by HumphreyYang** [comment]
Hi @jstac,

I have updated the links above according to your feedback. Should I send an email to Tom about dividing the DP squared lecture series?

Many thanks in advance.

**Reply by jstac** [reply]
Thanks @HumphreyYang and yes please --- just say that we might want to break it into multiple sections like the other lectures -- and if so what would be a good way to do it?  But I'm also happy for it to stay as is.

**Comment by jstac** [comment]
In each of the lectures, and same on existing live lectures, it says ", designed and written by [Thomas J. Sargent](http://www.tomsargent.com/) and [John Stachurski](http://johnstachurski.net/)."

Can we please replace this with just listing us as authors of the books?  I'm not sure what the default way to display authors of a JB is but presumably there is one?

On a related note, some of the opening sentences containing that phrase are not correct.  For example "This website presents a set of economic dynamics, designed and written by [Thomas J. Sargent](http://www.tomsargent.com/) and [John Stachurski](http://johnstachurski.net/)."

@mmcky , perhaps you can intervene here, fix these sentences to make them natural, as well as dropping the "designed and written by" part and just listing us as authors?

**Reply by jstac** [reply]
Noting this one too: "This website presents a set of equilibrium economic models, designed and written by [Thomas J. Sargent](http://www.tomsargent.com/) and [John Stachurski](http://johnstachurski.net/)."

We did not design and write the models, just the lectures.  (I should have provided more careful guidance to @HumphreyYang but I'm sure @mmcky will be able to rewrite these sentences naturally.)

**Comment by jstac** [comment]
This has not been done yet: 
> The lecture series https://quantecon.github.io/reorg.time-series-models/intro.html is too small. I suggest that we fold it into https://quantecon.github.io/reorg.tools-techniques/intro.html as a new section, titled "Time Series".

**Reply by HumphreyYang** [reply]
Many thanks @jstac,

It has been migrated to the tools and techniques. Apologize that I didn't take down the link.

**Reply by jstac** [reply]
Thanks @HumphreyYang , nice work.

**Reply by mmcky** [reply]
I have now deleted the `reorg.time-series-models` repo

**Comment by mmcky** [comment]
@HumphreyYang I reviewed the top level comment box with @jstac this morning and the meta data is now defined (except for descriptions). I will work my way through the projects to add a starting `description` in line with the comments above. 

Could you please:

- [x] update the repo names
- [x] update the titles

I will then:

- [ ] work on `descriptions`

**Reply by HumphreyYang** [reply]
Hi @mmcky,

I do not have the option to rename the repos on my end : ( 

Would you mind giving me the privilege or helping me to rename them?

Many thanks in advance.

**Reply by mmcky** [reply]
Roger that @HumphreyYang I have now renamed all repos, updated descriptions and tags, enabled gh-pages links on repo page, and updated all README.md files with repo name and title

**Reply by mmcky** [reply]
@HumphreyYang you will now need to update your scripts etc.

**Reply by HumphreyYang** [reply]
Many thanks @mmcky,

I will update everything together once I have the description for each lecture.

**Reply by mmcky** [reply]
@HumphreyYang please feel free to go ahead. 

We can easily update the descriptions as we review each series. We will change the design a little to add `author` under the title so there will be more changes coming down the pipeline anyway.

**Reply by HumphreyYang** [reply]
Many thanks @mmcky.

I will go ahead with these updates.

**Comment by mmcky** [comment]
Some starting points for **descriptions**:

1. `Statistics for Computational Economics`: This website presents a set of lectures on statistics for computational economics
2. `Dynamic Linear Economies`: This website presents a set of lectures on dynamic linear economies and tools needed for this this class of economic models
3. `Advanced Dynamic Programming`: This website presents a set of lectures on advanced topics in dynamic programming
4. `Equilibrium Models`: This website presents a set of lectures on equilibrium economic models
5. `Tools & Techniques for Computational Economics`: This website presents a set of lectures on the tools and techniques required to study computational economics

@HumphreyYang here are some adjusted descriptions. (cc @jstac)

Also it would be great if you could move the author to be above this description and using markdown links such as

```md
**Authors:** [Thomas J. Sargent](http://www.tomsargent.com/) and [John Stachurski](http://johnstachurski.net/)
```

which would render as

**Authors:** [Thomas J. Sargent](http://www.tomsargent.com/) and [John Stachurski](http://johnstachurski.net/)

This website presents ...

**Reply by HumphreyYang** [reply]
Hi @mmcky,

I think the description for the intro to dynamics is missing. Please let me know if I missed anything : )

Many thanks.

**Reply by HumphreyYang** [reply]
Also, please see the updated header for the table of content:
![Screenshot 2023-11-24 at 16 50 48](https://github.com/QuantEcon/meta/assets/39026988/ed248c63-17cb-4535-9aad-99074485f7dc)

Please let me know if there is anything I can further update.

**Reply by mmcky** [reply]
Thanks @HumphreyYang I think this looks good.

**Comment by mmcky** [comment]
For `Intro to Dynamics`

6. `Introduction to Economic Dynamics`: This website presents an introductory set of lectures on economic dynamics

cc: @HumphreyYang

**Comment by mmcky** [comment]
@jstac there is some work in progress on moving the authors up to be below the title https://github.com/QuantEcon/quantecon-book-theme/pull/239 but I think this looks good for the time being.

**Comment by jstac** [comment]
Roger that @mmcky .  Please ping me if you need me.

**Comment by mmcky** [comment]
@HumphreyYang can we link up sometime to review the `redirects` information. I think that is the last piece of the puzzle before we can look at migrating the lecture series over the new structure.

**Comment by mmcky** [comment]
@jstac @HumphreyYang I have been looking at `sphinx-reredirects` and it works really well and implements the most efficient redirect for specific pages (to external locations -- such as these new lecture locations). 

However using this example

https://657fd623738f65765c8108f8--nostalgic-wright-5fa355.netlify.app/intro.html

and scrolling down to `Heavy Tailed Distributions`

you will end up at the right lecture but on the `intro` lecture series (automatically re-directing you to the correct place). It might be a bit confusing for readers re: update in context with the only real clue is the url updated to intro.quantecon.org (from python.quantecon.org)

I am wondering if we need to figure out a way to prepend a small `notification` such as `This page has been moved to our new Introductory Quantitative Economics Lecture Series" (with a light coloured background) when connecting to that page from the redirect.

I don't have a technical solution for this yet but it might be a nice thing to be able to do.

**Reply by HumphreyYang** [reply]
Many thanks @mmcky for the example!

> I don't have a technical solution for this yet but it might be a nice thing to be able to do.

Perhaps we can have a redirect page saying you are heading to the new lecture series. Would it be easier to implement?

**Reply by mmcky** [reply]
> Perhaps we can have a redirect page saying you are heading to the new lecture series. Would it be easier to implement?

Do you mean a page that would display for 10seconds or so and then land you at the correct lecture? 

I think the `python.quantecon.org` lecture link will be pretty persistent out in the wild so I would prefer to get straight to the content page and give a context banner (if we can figure that bit out) -- that banner can let them know they need to update their links. 

Given our pages are static projects, I think it would need some `js` to show the banner (on the new site) if we can detect the connection is coming from a refresh meta forward. I will ask @DrDrij and @AakashGfude about this.

**Reply by mmcky** [reply]
The `primitive` alternative is to have `heavy_tails_redirect` which would simple be 

````

```{note}
information re redirect
```

```{include} heavy_tails.md
```

````

but will think of a more programmatic solution.

**Reply by mmcky** [reply]
Just an update on this. I have had discussions with the `EBP` community and we may build `sphinx-redirect-admonition`. It looks like there might be a way in javascript to detect a connection is coming from a referrer / redirect. I think this would be high value add if we can get this to work.

**Comment by mmcky** [comment]
@DrDrij would you be able to prepare Google codes for the new series and add them to the table in the top issue?

**Reply by DrDrij** [reply]
Onto it 👍

**Comment by mmcky** [comment]
@HumphreyYang how are you travelling with the internal link checking and resolution? Let me know if you need a second set of eyes to double check any patterns you've identified etc.

**Reply by HumphreyYang** [reply]
Many thanks @mmcky.

I put the link code into the code repo for lecture mapping: 
https://github.com/QuantEcon/lecture-mapping/blob/main/map_lectures.py#L330

It is not a robust solution given the diversity of `html` links and tags we have. What I will do next is to go through a manual check of the link found and create a `csv` file for link mapping.

I am focusing on reading DP book this week and will invest more time on lecture reorg after this week.

**Comment by mmcky** [comment]
@HumphreyYang can you let me know which new series will need `GPU` support (and hence `EC2` backend configuration).

**Reply by HumphreyYang** [reply]
Hi @mmcky,

I think only the stats series needs GPU support as it has the `bayes_nonconj` lecture.
